### ISSUE TITLE
Latest version doesn't allow you to join a singeplayer world on fabric

### ISSUE THREAD

**Issue by nullTheCoder** [ssblur/alchimiae#1]
mod version: 0.2.1-1.21

**What you expect to happen**:
Me being able to enter the world

**What actually happens**:
I get disconnected from a singleplayer world, with
```
[17:14:47] [Server thread/ERROR]: Couldn't place player in world
java.lang.NullPointerException: Cannot invoke "java.util.Collection.toArray()" because "c" is null
	at java.base/java.util.ArrayList.addAll(ArrayList.java:752) ~[?:?]
	at com.ssblur.alchimiae.data.IngredientEffectsSavedData.generate(IngredientEffectsSavedData.java:71) ~[alchimiae-fabric-0.2.1-1.21-sgd.jar:?]
	at com.ssblur.alchimiae.data.IngredientEffectsSavedData.<init>(IngredientEffectsSavedData.java:36) ~[alchimiae-fabric-0.2.1-1.21-sgd.jar:?]
	at net.minecraft.class_26.method_17924(class_26.java:52) ~[client-intermediary.jar:?]
	at com.ssblur.alchimiae.data.IngredientEffectsSavedData.computeIfAbsent(IngredientEffectsSavedData.java:128) ~[alchimiae-fabric-0.2.1-1.21-sgd.jar:?]
	at com.ssblur.alchimiae.data.IngredientMemorySavedData.fill(IngredientMemorySavedData.java:51) ~[alchimiae-fabric-0.2.1-1.21-sgd.jar:?]
	at com.ssblur.alchimiae.data.IngredientMemorySavedData.computeIfAbsent(IngredientMemorySavedData.java:148) ~[alchimiae-fabric-0.2.1-1.21-sgd.jar:?]
	at com.ssblur.alchimiae.data.IngredientMemorySavedData.computeIfAbsent(IngredientMemorySavedData.java:130) ~[alchimiae-fabric-0.2.1-1.21-sgd.jar:?]
	at com.ssblur.alchimiae.events.PlayerJoinedEvent.join(PlayerJoinedEvent.java:10) ~[alchimiae-fabric-0.2.1-1.21-sgd.jar:?]
	at net.minecraft.class_3324.handler$zbn000$architectury$placeNewPlayer(class_3324.java:1009) ~[client-intermediary.jar:?]
	at net.minecraft.class_3324.method_14570(class_3324.java:288) ~[client-intermediary.jar:?]
	at net.minecraft.class_8610.method_52408(class_8610.java:152) ~[client-intermediary.jar:?]
	at net.minecraft.class_8736.method_53043(class_8736.java:22) ~[client-intermediary.jar:?]
	at net.minecraft.class_8736.method_11054(class_8736.java:8) ~[client-intermediary.jar:?]
	at net.minecraft.class_2600.method_11072(class_2600.java:27) ~[client-intermediary.jar:?]
	at net.minecraft.class_3738.run(class_3738.java:18) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.method_18859(class_1255.java:162) ~[client-intermediary.jar:?]
	at net.minecraft.class_4093.method_18859(class_4093.java:23) ~[client-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_24306(MinecraftServer.java:864) ~[client-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_18859(MinecraftServer.java:173) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.method_16075(class_1255.java:136) ~[client-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_20415(MinecraftServer.java:846) ~[client-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_16075(MinecraftServer.java:840) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.method_18857(class_1255.java:145) ~[client-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_18857(MinecraftServer.java:810) ~[client-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_16208(MinecraftServer.java:815) ~[client-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_29741(MinecraftServer.java:702) ~[client-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:281) ~[client-intermediary.jar:?]
	at java.base/java.lang.Thread.run(Thread.java:1583) [?:?]
```
full log: https://mclo.gs/i504zfQ

**Comment by ssblur** [comment]
Apparently I accidentally fixed this issue at some point between releasing 0.2.1 and now, pushing changes. Thank you for reporting.

**Comment by ssblur** [comment]
This is fixed in 0.2.2, thank you for the report. Apparently there was a slight difference in the version built by GitHub and the version built on my computer.
